### PR TITLE
Update dash tests to use ingressgateway

### PR DESCRIPTION
**Pull request by douglas-reid**
The dashboard e2e tests were never updated to use ingressgateway. This
PR is an attempt to "modernize" the network config for those tests.

Fixes: https://github.com/istio/istio/issues/6164

### CODE REVIEW

**Comment by kyessenov** [comment]
Note that gateway (ingress and egress) has no mixer features enabled. Attempting to enable just the telemetry causes strange behavior (503s).

**Comment by douglas-reid** [comment]
@kyessenov yes. fortunately, these tests only need a way to get traffic to the fortio backend. ingress was consistently failing, so I'm trying to convert to ingressgateway. unfortunately, i can't get the local e2e tests to run on any of my machines. So, I guess I need to keep updating this PR in the meantime.

**Comment by douglas-reid** [comment]
and apparently, i can't get prow to run the dashboard tests anymore.

trying again:
/test e2e-suite-rbac-no_auth.sh

File: tests/e2e/tests/dashboard/fortio-rules.yaml [hunk 1]
```diff
@@ -9,39 +9,39 @@ spec:
...
-              servicePort: tcp-echo
+  selector:
+    istio: ingressgateway
+  servers:
+  - port:
+      number: 80
+      name: http
+      protocol: HTTP
+    hosts:
+    - echosrv
```
Comment (rshriram) [inline tests/e2e/tests/dashboard/fortio-rules.yaml]: wont work unless your client is literally supplying echosrv as the HTTP host.

Comment (douglas-reid) [reply tests/e2e/tests/dashboard/fortio-rules.yaml]: 👍 should I just remove the `hosts` section altogether?  I'm stuck just trying to get the tests to run, so I haven't had time to debug the setup.

**Comment by douglas-reid** [comment]
PTAL. Updated the network configs and ran the e2e locally. It passes on my minikube instance:

```
dougreid$ make e2e_dashboard E2E_ARGS="--use_local_cluster" HUB=localhost:5000 TAG=latest
...
2018-06-12T20:07:03.902105Z	info	Running test
=== RUN   TestDashboards
=== RUN   TestDashboards/Istio
=== RUN   TestDashboards/HTTP_Service
=== RUN   TestDashboards/TCP_Service
=== RUN   TestDashboards/Mixer
=== RUN   TestDashboards/Pilot
--- PASS: TestDashboards (0.82s)
	dashboard_test.go:78: Validating prometheus in ready-state...
	dashboard_test.go:478: Waiting for prometheus metrics: 0s
	dashboard_test.go:485: Value found for: 'round(sum(irate(istio_request_count[1m])), 0.001)'
	dashboard_test.go:485: Value found for: 'sum(irate(istio_request_count{response_code=~"4.*"}[1m]))'
	dashboard_test.go:485: Value found for: 'sum(irate(istio_request_count{response_code=~"5.*"}[1m]))'
	dashboard_test.go:83: Sentinel metrics found in prometheus.
    --- PASS: TestDashboards/Istio (0.08s)
    	dashboard_test.go:108: For Istio, checking 20 queries.
    --- PASS: TestDashboards/HTTP_Service (0.06s)
    	dashboard_test.go:108: For HTTP Service, checking 14 queries.
    --- PASS: TestDashboards/TCP_Service (0.08s)
    	dashboard_test.go:108: For TCP Service, checking 14 queries.
    --- PASS: TestDashboards/Mixer (0.36s)
    	dashboard_test.go:108: For Mixer, checking 45 queries.
    --- PASS: TestDashboards/Pilot (0.22s)
    	dashboard_test.go:108: For Pilot, checking 37 queries.
...
2018-06-12T20:08:07.372948Z	info	Namespace dashboard-test-6223ca3267c6437ea064b8 deletion status: true
2018-06-12T20:08:07.373320Z	info	Cleanup complete
ok  	istio.io/istio/tests/e2e/tests/dashboard	214.516s
```

**Comment by hklai** [comment]
FAIL	istio.io/istio/tests/e2e/tests/dashboard	33.130s

https://storage.googleapis.com/istio-prow/pull/istio_istio/6194/e2e-suite-rbac-no_auth/1779/build-log.txt

Still failing?

**Comment by douglas-reid** [comment]
@hklai those failures are the nodeport issue. When running locally, all the tests pass.

**Comment by douglas-reid** [comment]
One clue is that 33s is not long enough to run a test. This test and others are failing on prow because of test setup issues, and not tgr tests themselves.

**Comment by ldemailly** [comment]
we were supposed to not have readyness problem after fixit...

**Comment by douglas-reid** [comment]
@ldemailly if you are referring to the test setup stuff -- i think there is a bug in the cleanup of a prior test, leaving the nodePorts already bound. There is an issue open tracking that, so I expect resolution in the near-term.

**Pull request merged**